trusted_os/flash.go

```diff
@@ -142,46 +158,81 @@ func updateApplet(taELF []byte, taSig []byte, pb config.ProofBundle) (err error)
 			runtime.Gosched()
 			time.Sleep(100 * time.Millisecond)
 		}
-	}()
+	}
 
+	return blink, cancel
+}
+
+// updateApplet verifies an applet update and flashes it to internal storage
+func updateApplet(taELF []byte, taSig []byte, pb config.ProofBundle) (err error) {
 	log.Printf("SM applet verification")
 	if err = abconfig.Verify(taELF, taSig, PublicKey); err != nil {
 		return fmt.Errorf("applet verification error: %v", err)
 	}
 
+	return flashFirmware(Firmware_Applet, taELF, [][]byte{taSig}, pb)
+}
+
+// updateOS verifies an OS update and flashes it to internal storage
+func updateOS(osELF []byte, osSigs [][]byte, pb config.ProofBundle) (err error) {
+	// TODO: OS signature verification
+
+	return flashFirmware(Firmware_OS, osELF, osSigs, pb)
+}
+
+// flashFirmware writes config & elf bytes to the MMC in the correct region for the specificed type of firmware.
+func flashFirmware(t FirmwareType, elf []byte, sigs [][]byte, pb config.ProofBundle) error {
+	blink, cancel := blinkenLights()
+	defer cancel()
+	go blink()
+
+	var (
+		confBlock int
+		elfBlock  int
+	)
```

Comment:
I'm not alone in my madness - there are examples of this in the stdlib too: https://cs.opensource.google/go/go/+/master:src/reflect/value.go;l=2475?q=%22var%20(%22&ss=go%2Fgo&start=11 :)

On the one hand it's 4 lines when it could be 2, on the other it's less cluttered than `vars` on every line... hmm.

Reply:
I've replaced with `:=` instead since they're just `int`s.